superset/migrations/versions/2023-07-18_15-30_863adcf72773_delete_obsolete_druid_nosql_slice_parameters.py

```diff
@@ -0,0 +1,102 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""delete obsolete Druid NoSQL slice parameters
+
+Revision ID: 863adcf72773
+Revises: 6d05b0a70c89
+Create Date: 2023-07-18 15:30:43.695135
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "863adcf72773"
+down_revision = "6d05b0a70c89"
+
+import json
+
+from alembic import op
+from sqlalchemy import Column, Integer, Text
+from sqlalchemy.ext.declarative import declarative_base
+
+from superset import db
+
+Base = declarative_base()
+
+
+class Slice(Base):
+    __tablename__ = "slices"
+
+    id = Column(Integer, primary_key=True)
+    params = Column(Text)
+    query_context = Column(Text)
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+
+    for slc in session.query(Slice).all():
+        if slc.params:
+            updated = False
+
+            try:
+                params = json.loads(slc.params)
+
+                for key in ["druid_time_origin", "granularity"]:
+                    if key in params:
+                        del params[key]
+                        updated = True
+
+                if updated:
+                    slc.params = json.dumps(params)
+            except Exception:
+                pass
```

Comment:
can we add a log here to see the exception

Reply:
@hughhhh good idea.